packages-ts/gauntlet-terra/src/commands/addressBook.ts

```diff
@@ -0,0 +1,95 @@
+import { logger } from '@chainlink/gauntlet-core/dist/utils'
+import { assertions } from '@chainlink/gauntlet-core/dist/utils'
+import { assert } from 'console'
+import { ContractId } from './types'
+
+type Instance = {
+  name: string
+  contractId: ContractId
+}
+
+class AddressBook {
+  operator: string
+  instances: Map<string, Instance> // address => instance name
+
+  constructor() {
+    this.instances = new Map<string, Instance>()
+  }
+
+  setOperator(address: string) {
+    this.operator = address
+  }
+  addInstance(contractId: ContractId, address: string, name?: string) {
+    this.instances.set(address, { name, contractId } as Instance)
+    logger.debug(`Using deployed instance of ${contractId}: ${name}=${address}`)
+    return this
+  }
+
+  // format():  Automatically format terra addresses depending on contract type.
+  //
+  // Note:
+  //  use(withAddressBook) middleware must be enabled for any commands calling this
+  //
+  // Example use:
+  //  import addressBook as ab from '@chainlink/gauntlet-terra'
+  //
+  //  Use ${ab.format(address)} instead of ${address} in strings sent to console or log.
+  //   - If it matches the address added with name='multisig', the address will show up
+  //     as yellow and labelled "multisig".
+  //   - If it matches a known contract address read from the environemnt (LINK, BILLING_ACCESS_CONTROLLER,... ),
+  //     the address will be blue and labelled with the contract name( or "name" if specified )
+  //   - Unknown addresses will remain unformmated
+  format(address: string): string {
```

Comment:
Is not `AddressBook` responsibility to format or print anything, just store known contracts. 
It should be `logger` responsibility to format an address if it's known. 
For now, we can wrap our logger into a Terra logger until `gauntlet-core` logger can accept some address book. 
A very simple version could be similar to this:
```js
import { logger } from '@chainlink/gauntlet-core/dist/utils'

class AddressBook {
  contracts: Record<string, any>

  constructor() {
    this.contracts = {}
  }

  addContract = (address, name) => {
    this.contracts[address] = name
  }
}

const TerraLogger = () => {
  let addressBook: AddressBook
  const newLogger = {
    ...logger,
    withAddressBook: (book: AddressBook) => {
      addressBook = book
    },
    formatAddress: (address) => {
      if (!addressBook) return address
      if (addressBook.contracts[address]) {
        // Apply styles
        return logger.style(`${addressBook.contracts[address]} (${address})`, 'yellow', 'dim')
      }
      return address
    },
  }
  return newLogger
}

// Initialize our wrapped logger
const myLogger = TerraLogger()

// Initialize the new book. Probably in the middleware
const myTerraBook = new AddressBook()

// Include the book into the logger
myLogger.withAddressBook(myTerraBook)

// We can add contracts to our book
myTerraBook.addContract('my_address', 'MY_KNOWN_ADDRESS')

// Example of usage
myLogger.info(`This is my message with known address ${myLogger.formatAddress('my_address')}`)
myLogger.info(`This is my message with unknown address ${myLogger.formatAddress('some_address')}`)
```

Reply:
Alright, I implemented something like this in my latest commit.  There is still a minor issue to be settled about where exactly to store the address book and how many to have (one global book, or one local book for each command).  I changed it to be per-command in this commit, but global seems preferable to me so I included my analysis of that in the commit message.